### PR TITLE
[WIP] AzurePerformanceDataUtility 

### PR DESCRIPTION
**Pull request by johannes-manner**
PR for implementing the azure performance utitilty. A fetcher for azure log files.
Refs #1

### CODE REVIEW

File: src/main/java/de/uniba/dsg/serverless/azure/AzureLogAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package de.uniba.dsg.serverless.azure;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AzureLogAnalyzer {
+
+	public static final String REQUEST_ID_REGEX = "Id=[0-9a-f\\-]+";
```
Comment (johannes-manner) [inline src/main/java/de/uniba/dsg/serverless/azure/AzureLogAnalyzer.java]: Make the static fields private, if not used in other classes.

File: src/main/java/de/uniba/dsg/serverless/azure/AzureLogHandler.java [hunk 2]
```diff
@@ -1,6 +1,164 @@
...
+					bw.write(performanceData.toCSVString() + System.lineSeparator());
+				}
+			}
+
+		} catch (SeMoDeException e) {
+			logger.severe(e.getMessage());
+			logger.severe("Data handler is terminated due to an error.");
+		} catch (IOException e) {
+			logger.severe("IO Exception occured.");
+			e.printStackTrace();
```
Comment (johannes-manner) [inline src/main/java/de/uniba/dsg/serverless/azure/AzureLogHandler.java]: Remove stack traces.

File: src/main/java/de/uniba/dsg/serverless/azure/AzureLogHandler.java [hunk 3]
```diff
@@ -1,6 +1,164 @@
...
+				LocalDateTime startTime = AzureLogAnalyzer.parseTime(lineParts[0]);
+				pendingData.put(id, startTime);
+			} else if (message.startsWith(FUNCTION_COMPLETED)) {
+				// Finish Message
+				String id = AzureLogAnalyzer.extractRequestId(message);
+				LocalDateTime startTime = pendingData.remove(id);
+				double duration = AzureLogAnalyzer.extractDuration(message);
+				PerformanceData data = new PerformanceData(functionName, "", id, startTime, duration, -1, -1, -1);
+				result.add(data);
+			} else {
```
Comment (johannes-manner) [inline src/main/java/de/uniba/dsg/serverless/azure/AzureLogHandler.java]: Removing else block. Not necessary here.

File: src/main/java/de/uniba/dsg/serverless/azure/AzureLogHandler.java [hunk 4]
```diff
@@ -1,6 +1,164 @@
...
+
 public class AzureLogHandler {
+
+	private static final String FUNCTION_COMPLETED = "Function completed";
+
+	private static final String FUNCTION_STARTED = "Function started";
+
+	private static final Logger logger = Logger.getLogger(AzureLogHandler.class.getName());
+
+	public final String storageConnectionString;
```
Comment (johannes-manner) [inline src/main/java/de/uniba/dsg/serverless/azure/AzureLogHandler.java]: Make fields private, especially the connection string is a potential security issue. Fields are not used in other classes, therefore no access is needed.

File: src/main/java/de/uniba/dsg/serverless/model/PerformanceData.java [hunk 5]
```diff
@@ -30,7 +30,7 @@ public PerformanceData() {
 		this.memoryUsed = -1;
 	}
 
-	public PerformanceData(String functionName, String logStream, String requestId, long startTime, 
+	public PerformanceData(String functionName, String logStream, String requestId, long startTime,
```
Comment (johannes-manner) [inline src/main/java/de/uniba/dsg/serverless/model/PerformanceData.java]: Two constructors are not necessary here. Please refactor this constructor and change the constructor call in AWSLogAnalyzer line 232. LocalDateTime startTime is the better option in this case.

**Comment by johannes-manner** [comment]
Also update the README.md file with the new option 👍

File: src/main/java/de/uniba/dsg/serverless/util/AzurePerformanceDataUtility.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+	public void start(List<String> args) {
+
+		if (args.size() < 4) {
+			AzurePerformanceDataUtility.logger.log(Level.SEVERE, "Wrong parameter size: " + "\n(1) Account Name "
+					+ "\n(2) Account Key " + "\n(3) Share" + "n(4) Function Name");
+			return;
+		}
+
+		String accountName = args.get(0);
```
Comment (johannes-manner) [inline src/main/java/de/uniba/dsg/serverless/util/AzurePerformanceDataUtility.java]: To be more precise, it's the storageAccountName. Isn't it?

File: src/main/java/de/uniba/dsg/serverless/util/AzurePerformanceDataUtility.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+		String accountKey = args.get(1);
+		String shareName = args.get(2);
```
Comment (johannes-manner) [inline src/main/java/de/uniba/dsg/serverless/util/AzurePerformanceDataUtility.java]: Isn't it the same as the accountName above?

File: src/main/java/de/uniba/dsg/serverless/util/AzurePerformanceDataUtility.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+	public void start(List<String> args) {
+
+		if (args.size() < 4) {
+			AzurePerformanceDataUtility.logger.log(Level.SEVERE, "Wrong parameter size: " + "\n(1) Account Name "
+					+ "\n(2) Account Key " + "\n(3) Share" + "n(4) Function Name");
+			return;
+		}
+
+		String accountName = args.get(0);
+		String accountKey = args.get(1);
```
Comment (johannes-manner) [inline src/main/java/de/uniba/dsg/serverless/util/AzurePerformanceDataUtility.java]: Put emphasis on using key1 or is the execution also possible with key2 in the storage service?

File: src/main/java/de/uniba/dsg/serverless/util/AzurePerformanceDataUtility.java (hunk 7 above)
Comment (martin-endress) [reply src/main/java/de/uniba/dsg/serverless/util/AzurePerformanceDataUtility.java]: In most cases, yes. The storage account is not necessarily the same. (defined when creating the storage account)

**Pull request merged**